### ISSUE TITLE
TypeError: Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by 0x77dev** [aholstenson/ataraxia#2]
### Error
```shell
$ env DEBUG="*" node index.js
  ataraxia About to join network as 1z7eIvirsA +0ms
  ataraxia:local Starting with id `1z7eIvirsA` +0ms
  local-machine-network Connecting as client +0ms
  local-machine-network Connected to leader +2ms
  ataraxia:local:peer Sending hello with data { id: '1z7eIvirsA', version: 2 } +0ms
(node:15523) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.getOwnPropertyNames (<anonymous>)
    at LocalService.get definition [as definition] (/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/service-local.js:30:27)
    at Services.register (/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/index.js:117:72)
    at /Users/0x77/Projects/lit79/core/index.js:21:26
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:15523) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:15523) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
  ataraxia:local:peer Incoming hello with payload { id: '1z7e5hMxP8', version: 2 } +33ms
  ataraxia:local:peer:1z7e5hMxP8 Sending ping with data undefined +0ms
  ataraxia:local:peer:1z7e5hMxP8 Incoming ping with payload null +0ms
  ataraxia:topology Node 1z7e5hMxP8 now reachable +0ms
  ataraxia:services Requesting initial list of services from Node{1z7e5hMxP8} +0ms
  ataraxia:local:peer:1z7e5hMxP8 Sending message with data [
  '1z7eIvirsA',
  '1z7e5hMxP8',
  { type: 'service:list', data: { lastVersion: 0 } }
] +2ms
  ataraxia:local:peer:1z7e5hMxP8 Incoming message with payload [
  '1z7e5hMxP8',
  '1z7eIvirsA',
  { type: 'service:list', data: { lastVersion: 0 } }
] +10ms
/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/service-local.js:30
                for(const key of Object.getOwnPropertyNames(this.instance.metadata)) {
                                        ^

TypeError: Cannot convert undefined or null to object
    at Function.getOwnPropertyNames (<anonymous>)
    at LocalService.get definition [as definition] (/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/service-local.js:30:27)
    at Services._handleServiceList (/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/index.js:212:27)
    at Services._handleMessage (/Users/0x77/Projects/lit79/core/node_modules/ataraxia-services/index.js:164:10)
    at EventEmitter.emit (events.js:210:5)
    at EventEmitter.emit (domain.js:478:20)
    at EventEmitter.<anonymous> (/Users/0x77/Projects/lit79/core/node_modules/ataraxia/index.js:81:17)
    at EventEmitter.emit (events.js:210:5)
    at EventEmitter.emit (domain.js:478:20)
    at Topology.handleMessage (/Users/0x77/Projects/lit79/core/node_modules/ataraxia/topology/index.js:189:16)
```
### Code
```javascript
const Network = require('ataraxia');
const LocalTransport = require('ataraxia-local');
const TCPTransport = require('ataraxia-tcp');
const Services = require('ataraxia-services');
const cli = require("repl").start("> ");

const net = new Network({ name: '302' });

const local = new LocalTransport();
local.on('leader', () => {
	net.addTransport(new TCPTransport());
});
net.addTransport(local);

net.start().then(() => {
	const services = new Services(net);
	global.services = services;
	cli.context.services = services;
	services.on('available', service => console.log(service.id, 'is now available'));
	services.on('unavailable', service => console.log(service.id, 'is no longer available'));
	const handle = services.register('service-id', {
		hello() {
			return 'Hello world';
		}
	});

	// Emit events
	handle.emitEvent('hello', { data: 'goes-here' });

	// Interact with services
	const service = services.get('service-id');
	if (service) {
		service.hello()
			.then(result => console.log('service said', result))
			.catch(handleErrorCorrectlyHere);

		service.on('hello', data => console.log('got', data));
	}
});
cli.context.net = net;
```
### Node.js Version
```shell
$ node -v
v13.2.0
```
### package.json
```json
{
  "dependencies": {
    "ataraxia": "^0.7.2",
    "ataraxia-local": "^0.7.2",
    "ataraxia-services": "^0.7.2",
    "ataraxia-tcp": "^0.7.2"
  }
}
```

**Comment by aholstenson** [comment]
Thanks for reporting this! Seems like I was lazy and services requires a `metadata` property to work. I'll probably look at and release a fix for this in the next few days.

**Comment by 0x77dev** [comment]
I will wait for good news, thanks)

**Comment by aholstenson** [comment]
Just published `ataraxia-services` version 0.7.3 where this issue should be fixed. Thanks again for reporting it!
